keanu-project/src/main/java/io/improbable/keanu/algorithms/variational/optimizer/Optimizer.java

```diff
@@ -100,40 +106,57 @@ static Optimizer ofConnectedGraph(Vertex<?> vertexFromNetwork) {
         int position = 0;
         double[] point = new double[(int) totalLatentDimensions];
 
-        for (Vertex<? extends NumberTensor> vertex : continuousLatentVertices) {
-            double[] values = vertex.getValue().asFlatDoubleArray();
+        for (String variable : latentVariables) {
+            double[] values = latentVariableValues.get(variable).asFlatDoubleArray();
             System.arraycopy(values, 0, point, position, values.length);
             position += values.length;
         }
 
         return point;
     }
 
-    static void setAndCascadePoint(double[] point, List<? extends Vertex<DoubleTensor>> latentVertices) {
+    static Map<String, DoubleTensor> convertFromPoint(double[] point,
+                                                      List<String> latentVariables,
```

Comment:
you don't need the 2nd argument, `latentVariables`, because you have the same information in the keys of the 3rd argument

Reply:
👍

Reply:
The list is needed because the order needs to be preserved.